drivers/wifi/simplelink.c

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2018 Texas Instruments, Incorporated
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_WIFI_LEVEL
+#define SYS_LOG_DOMAIN "dev/simplelink"
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <device.h>
+#include <net/net_if.h>
+#include <net/wifi_mgmt.h>
+#include <net/net_offload.h>
+
+#include "simplelink_support.h"
+
+struct simplelink_data {
+	struct net_if *iface;
+	unsigned char mac[6];
+};
+
+static struct simplelink_data simplelink_data;
+
+/* Handle connection events from the SimpleLink Event Handlers: */
+static void simplelink_wifi_cb(u32_t mgmt_event, struct sl_connect_state *conn)
```

Comment:
Use your own code, not mgmt_event, to tell your callback what it should do here.

You driver is not listening to NET_EVENT_WIFI_CMD_(DIS)CONNECT_RESULT, nor NET_EVENT_IPV4_ADDR_ADD (not to say about the fact you are mixing up event cmd enums and macros here.)

Reply:
OK, good point.

Reply:
rename mgmt_event to event.

Reply:
OK, done.